services/app-web/src/common-code/domain-models/submission.ts

```diff
@@ -135,6 +136,64 @@ function submissionName(
     return `MCR-${submission.stateCode.toUpperCase()}-${formattedProgramNames}-${padNumber}`
 }
 
+export type RateDataType = {
+    rateType?: 'AMENDMENT' | 'NEW' | null
+    rateDateStart?: Date
+    rateDateEnd?: Date
+    rateDateCertified?: Date
+    rateAmendmentInfo?: {
+        effectiveDateEnd?: Date
+        effectiveDateStart?: Date
+    } | null
+}
+
+export const generateRateName = (
```

Comment:
This would be a good function to write some table tests for. Take a look at the "Submission name is correct" test in submission.test.ts